LuaUI/Widgets/cmd_hold_ctrl_to_place_and_morph.lua

```diff
@@ -0,0 +1,126 @@
+local widgetName = "Hold Ctrl during placement to morph";
+
+function widget:GetInfo()
+	return {
+		name      = widgetName,
+		version   = '1.0',
+		desc      = "Hold the Ctrl key while placing a Cornea, Aegis, Radar Tower or Geothermal Reactor to issue a morph order to the nanoframe when it's created (which will make it start morphing once finished).",
+		author    = "dunno",
+		date      = "2022-05-18",
+		license   = "MIT",
+		layer     = 0,
+		enabled   = true
+	}
+end
+
+---@alias UnitInternalName string
+---@alias UnitDefId integer
+---@alias UnitId integer
+
+---@param names UnitInternalName[]
+---@return table<UnitDefId, boolean>
+local function CreateUnitDefIdSet(names)
+	local result = {}
+	for _, name in ipairs(names) do
```

Comment:
Avoid `ipairs`, it's slow compared to regular iteration (in this case it doesn't really matter since it's just 4 entries in the table but maybe someone else would copy-paste code in the future):
```suggestion
	for i = 1, #names do
		local name = names[i]
```